### DISCUSSION TITLE
Need help tunneling WireGuard

### DISCUSSION THREAD

**Discussion by poqdavid** [shadowsocks/shadowsocks-rust#1203] [Q&A]
I am trying to tunnel WireGuard using shadowsocks but I can't make a handshake to the WireGuard at all I wonder how I can fix it 
Also shadowsocks and WireGuard are hosted on different servers and we use Ubuntu with firewalld

This is my shadowsocks server config
```json
{
  "server": "0.0.0.0",
  "server_port": 8388,
  "password": "<password>",
  "timeout": 300,
  "method": "chacha20-ietf-poly1305",
  "mode": "tcp_and_udp"
}
```

and here my config for the sslocal.exe

```json
{
    "server":"<IP VPS1>",
    "server_port":8388,
    "password":"<password>",
    "method":"chacha20-ietf-poly1305",
    "mode": "tcp_and_udp",
    "locals": [
        {
            "mode":"tcp_and_udp",
            "protocol":"tunnel",
            "forward_address":"<IP VPS2>",
            "forward_port":31194,
            "local_address":"127.0.0.3",
            "local_port":31193
        },
        {
            "mode":"tcp_and_udp",
            "protocol":"tunnel",
            "forward_address":"<IP VPS1>",
            "forward_port":2222,
            "local_address":"127.0.0.3",
            "local_port":22
        }
    ]
}
```

When I tested I could use the tunnel and SSH to the server but WireGuard didn't connect

**Comment by vvyushmanov** [comment]
Hello!

This most probably happens because once WireGuard starts, SS traffic switches to this new interface, creating a kind of loop.
There's a config option "outbound_bind_interface" available, but I can't figure out how to make it work on Windows (probably will report it as a bug).

As a workaround, you can add a static route in Windows. Open PowerShell as admin and type:

`route -p add <server_ip> MASK 255.255.255.255 <default_gateway_ip>`

You can add adresses one by one or add a whole subnet by tweaking the subnet mask.
